### DISCUSSION TITLE
additional OBIS codes

### DISCUSSION THREAD

**Discussion by antonmeyer** [scs/smartmeter-datacollector#52] [General]
I've added some OBIS codes.
I missed the point, why to handle that in 2 different files.
And why this should not be part of the Meter definition.
In general the M-Bus should be the same, and than the presented OBIS data depends on the meter-type and / or the meter operator (Messstellenbetreiber / Stadtwerk)
Would be perfect to have a list in one file per meter-type, may be as part of the .ini file?
I also did not get the reason why OBISCode(1, 1, 5, 8, 1) also matches. (lack of knowledge)

REACTIVE_ENERGY_Q1_T1 = MeterDataPointType("REACTIVE_ENERGY_Q1_T1", "Reactive Energy +Ri Q1 - Tariff 1", "VAh")
    REACTIVE_ENERGY_Q1_T2 = MeterDataPointType("REACTIVE_ENERGY_Q1_T2", "Reactive Energy +Ri Q1 - Tariff 2", "VAh")
    REACTIVE_ENERGY_Q2_T1 = MeterDataPointType("REACTIVE_ENERGY_Q2_T1", "Reactive Energy +Rc Q2 - Tariff 1", "VAh")
    REACTIVE_ENERGY_Q2_T2 = MeterDataPointType("REACTIVE_ENERGY_Q2_T2", "Reactive Energy +Rc Q2 - Tariff 2", "VAh")
    REACTIVE_ENERGY_Q3_T1 = MeterDataPointType("REACTIVE_ENERGY_Q3_T1", "Reactive Energy -Ri Q3 - Tariff 1", "VAh")
    REACTIVE_ENERGY_Q3_T2 = MeterDataPointType("REACTIVE_ENERGY_Q3_T2", "Reactive Energy -Ri Q3 - Tariff 2", "VAh")
    REACTIVE_ENERGY_Q4_T1 = MeterDataPointType("REACTIVE_ENERGY_Q4_T1", "Reactive Energy -Rc Q4 - Tariff 1", "VAh")
    REACTIVE_ENERGY_Q4_T2 = MeterDataPointType("REACTIVE_ENERGY_Q4_T2", "Reactive Energy -Rc Q4 - Tariff 2", "VAh")
    RegisterCosem(OBISCode(1, 0, 5, 8, 1), MeterDataPointTypes.REACTIVE_ENERGY_Q1_T1.value),
    RegisterCosem(OBISCode(1, 0, 5, 8, 2), MeterDataPointTypes.REACTIVE_ENERGY_Q1_T2.value),
    RegisterCosem(OBISCode(1, 0, 6, 8, 1), MeterDataPointTypes.REACTIVE_ENERGY_Q2_T1.value),
    RegisterCosem(OBISCode(1, 0, 6, 8, 2), MeterDataPointTypes.REACTIVE_ENERGY_Q2_T2.value),
    RegisterCosem(OBISCode(1, 0, 7, 8, 1), MeterDataPointTypes.REACTIVE_ENERGY_Q3_T1.value),
    RegisterCosem(OBISCode(1, 0, 7, 8, 2), MeterDataPointTypes.REACTIVE_ENERGY_Q3_T2.value),
    RegisterCosem(OBISCode(1, 0, 8, 8, 1), MeterDataPointTypes.REACTIVE_ENERGY_Q4_T1.value),
    RegisterCosem(OBISCode(1, 0, 8, 8, 2), MeterDataPointTypes.REACTIVE_ENERGY_Q4_T2.value),

**Comment by raymar9** [comment]
Hi antonmeyer

> I missed the point, why to handle that in 2 different files.
In one file we define all the `MeterDataPointTypes`. These are the common data types which are used to define data forwarded to the different data sinks (e.g. MQTT) and shall be the same for all attached meters.
In the second location we map the DLMS/COSEM world (by the OBIS codes) to the `MeterDataPointTypes`.

> And why this should not be part of the Meter definition.
> In general the M-Bus should be the same, and than the presented OBIS data depends on the meter-type and / or the meter operator (Messstellenbetreiber / Stadtwerk)
> Would be perfect to have a list in one file per meter-type, may be as part of the .ini file?

Only lately, we restructured the whole OBIS detection / parsing scheme. Before, we had a list of OBIS-MeterDataPointTypes Mapping for each meter. However, we saw that not only the different meter manufacturer but also the meter operators use different DLMS configuration / different OBIS codes (like different channel). Therefore, this does not scale and we changed to the current implementation. Now, only the relevant part of the OBIS codes (positions A, C, D, E) are compared and matched with the OBIS from one default mapping table. This is why OBISCode(1, 1, 5, 8, 1) matches as well. Now, the same list can be well used for different meters and operators.
